graylog2-web-interface/src/components/common/SearchForm.jsx

```diff
@@ -88,27 +103,37 @@ class SearchForm extends React.Component {
   _onSearch = (e) => {
     e.preventDefault();
 
-    this._setLoadingState();
-    this.props.onSearch(this.refs.query.value, this._resetLoadingState);
+    this._setLoadingState().then(() => {
+      this.props.onSearch(this.state.query, this._resetLoadingState);
+    });
   };
 
   _onReset = () => {
     this._resetLoadingState();
-    this.refs.query.value = '';
-    this.props.onReset();
+    this.setState({ query: '' }, this.props.onReset);
```

Comment:
As `this.state.query` is initialised with the value from `this.props.query`, I guess resetting the value should also use `this.props.query`, shouldn't it? If no query was given as prop, in the end it would also use `''` as default.

Reply:
Resetting the query should also call `this.props.onQueryChange()`.